pkg/controller/job/job_controller.go

```diff
@@ -1512,8 +1512,19 @@ func activePodsForRemoval(job *batch.Job, pods []*v1.Pod, rmAtLeast int) []*v1.P
 	}
 
 	if len(rm) < rmAtLeast {
-		sort.Sort(controller.ActivePods(left))
-		rm = append(rm, left[:rmAtLeast-len(rm)]...)
+		numToDelete := rmAtLeast - len(rm)
```

Comment:
@ahg-g is this controller change required?

Reply:
No, we shouldn't need this. This behavior is already implemented via https://github.com/kubernetes/kubernetes/blob/a58e8993922ca0cee85a9e982f66ea5c9039ec93/pkg/controller/job/indexed_job_utils.go#L229

@danielvegamyhre can you please add integration tests to verify the behavior without any changes to the job controller.